Tests/OpenFeatureTests/DeveloperExperienceTests.swift

```diff
@@ -19,6 +19,38 @@ final class DeveloperExperienceTests: XCTestCase {
         XCTAssertFalse(flagValue)
     }
 
+    func testObserveGlobalEvents() {
+        let readyExpectation = XCTestExpectation(description: "Ready")
+        let errorExpectation = XCTestExpectation(description: "Error")
+        let staleExpectation = XCTestExpectation(description: "Stale")
+        var eventState = OpenFeatureAPI.shared.observe().sink { event in
+            switch event {
+            case ProviderEvent.ready:
+                readyExpectation.fulfill()
+            case ProviderEvent.error:
+                errorExpectation.fulfill()
+            case ProviderEvent.stale:
+                staleExpectation.fulfill()
+            default:
+                XCTFail("Unexpected event")
+            }
+        }
+        let provider = DoSomethingProvider()
+        OpenFeatureAPI.shared.setProvider(provider: provider)
+        wait(for: [readyExpectation], timeout: 5)
+
+        OpenFeatureAPI.shared.clearProvider()
+
+        // Clearing the Provider shouldn't send further global events from it
+        eventState = OpenFeatureAPI.shared.observe().sink { _ in
+            XCTFail("Unexpected event")
+        }
```

Comment:
I suggest to use: 
`observe().dropFirst().sink { ... }`
And keep the line `OpenFeatureAPI.shared.clearProvider()` after the observe call